data-exporter/src/main/java/org/reactome/release/dataexport/UniProtReactomeEntry.java

```diff
@@ -127,104 +160,140 @@ private UniProtReactomeEntry(long dbId, String accession, String displayName) {
+	 * as the UniProt entry in Reactome is represented by the returned map's keys rather than only the UniProt accession
+	 */
+	@Deprecated
+	public static Map<String, Set<ReactomeEvent>> fetchUniProtAccessionToReactomeEvents(
+		Session graphDBSession
+	) {
+		return fetchUniProtReactomeEntryToReactomeEvents(graphDBSession)
+			.entrySet()
+			.stream()
+			.collect(
+				Collectors.toMap(
+					entry -> entry.getKey().getAccession(),
+					Map.Entry::getValue
+				)
+			);
+	}
+
+	/**
+	 * Retrieves, from the graph database, a Map of UniProtReactomeEntry objects to the set of events (both Pathways
+	 * and Reaction Like Events) in which each UniProtReactomeEntry participates
 	 * @param graphDBSession Neo4J Driver Session object for querying the graph database
-	 * @return Map of UniProt accession to set of Reactome Events in Reactome
+	 * @return Map of UniProtReactomeEntry objects to set of Reactome Events in Reactome
 	 */
-	public static Map<String, Set<ReactomeEvent>> fetchUniProtAccessionToReactomeEvents(Session graphDBSession) {
-		if (uniprotAccessionToReactomeEventCache.containsKey(graphDBSession)) {
-			return uniprotAccessionToReactomeEventCache.get(graphDBSession);
+	public static Map<UniProtReactomeEntry, Set<ReactomeEvent>> fetchUniProtReactomeEntryToReactomeEvents(
+		Session graphDBSession
+	) {
+		if (uniprotReactomeEntryToReactomeEventCache.containsKey(graphDBSession)) {
+			return uniprotReactomeEntryToReactomeEventCache.get(graphDBSession);
 		}
 
 		logger.info("Computing UniProt to Reactome events");
 
 		Map<Long, ReactomeEvent> eventCache = ReactomeEvent.fetchReactomeEventMap(graphDBSession);
-		Map<String, Set<Long>> uniprotAccessionToReactionLikeEventId = fetchUniProtAccessionToRLEId(graphDBSession);
+		Map<UniProtReactomeEntry, Set<Long>> uniprotReactomeEntryToReactionLikeEventId =
+			fetchUniProtReactomeEntryToRLEId(graphDBSession);
 		Map<Long, Set<Long>> rleToPathwayId = PathwayHierarchyUtilities.fetchRLEIdToPathwayId(graphDBSession);
 
 		AtomicInteger count = new AtomicInteger(0);
-		Map<String, Set<ReactomeEvent>> uniprotAccessionToReactomeEvent = new ConcurrentHashMap<>();
-		uniprotAccessionToReactionLikeEventId.keySet().stream().forEach(uniprotAccession -> {
+		Map<UniProtReactomeEntry, Set<ReactomeEvent>> uniprotReactomeEntryToReactomeEvent = new ConcurrentHashMap<>();
+		uniprotReactomeEntryToReactionLikeEventId.keySet().stream().forEach(uniprotAccession -> {
 			Set<Long> reactionLikeEventIds =
-				uniprotAccessionToReactionLikeEventId.computeIfAbsent(uniprotAccession, k -> new HashSet<>());
+				uniprotReactomeEntryToReactionLikeEventId.computeIfAbsent(uniprotAccession, k -> new HashSet<>());
 
 			// Add RLEs containing UniProt accession
-			uniprotAccessionToReactomeEvent
+			uniprotReactomeEntryToReactomeEvent
 				.computeIfAbsent(uniprotAccession, k -> new HashSet<>())
 				.addAll(reactionLikeEventIds.stream().map(eventCache::get).collect(Collectors.toSet()));
```

Comment:
This expression (and others like it) is rather dense, could probably use a comment to explain what is going on here.

Reply:
That's fair - I'll work on making it more readable.